src/main/scala/models/Domain.scala

```diff
@@ -0,0 +1,30 @@
+package models
+
+import io.circe.{Codec, Decoder, Encoder}
+import io.circe.generic.semiauto.deriveCodec
+
+object Domain {
+
+  final case class CustomerId(value: String)
+
+  object CustomerId {
+    implicit val encodeCustomerId: Encoder[CustomerId] = Encoder.encodeString.contramap[CustomerId](_.value)
+    implicit val decodeCustomerId: Decoder[CustomerId] = Decoder.decodeString.map(CustomerId(_))
+    implicit val customerIdCodec: Codec[CustomerId]    = Codec.from(decodeCustomerId, encodeCustomerId)
+  }
+
+  final case class CustomerName(value: String)
```

Comment:
that things stay here, but in `gateway` u have `CustomerApiOutput.fromDomain(customer) = CustomerApiOutput(customer.id.value)...`

and the same for `input`: `CustomerApiInput -> Customer`